pbs/pbsrequest.go

```diff
@@ -162,6 +162,8 @@ func ParsePBSRequest(r *http.Request, cache cache.Cache) (*PBSRequest, error) {
 		if err != nil {
 			return nil, fmt.Errorf("Invalid URL '%s': %v", url.Host, err)
 		}
+	} else {
+		pbsReq.SyncMap = NewSyncMap()
```

Comment:
The short answer is... I'm not sure. I definitely agree that it's confusing.

On the other hand... `func (req *PBSRequest) GetUserID(BidderCode string)` surfaced some interesting unit test failures. Apparently in Go any reads on a `nil` map are treated as if that map were empty. So `pbsReq.UIDs['key']` will just return nil if `UIDs` is undefined, but `req.SyncMap.GetUID()` will cause panic. Stashing an empty `SyncMap` on here makes reads behave nicely.

I think this lends some extra weight to your suggestion [here](https://github.com/prebid/prebid-server/pull/86#discussion_r134610843). If the methods are defined on a struct, then I can define how they behave on a `nil`. Interfaces don't give that flexibility.